File: app/src/main/java/com/greenwald/aaron/ridetracker/model/Segment.java
package com.greenwald.aaron.ridetracker.model;

import java.util.ArrayList;
import java.util.Date;

public class Segment {
    private Date startedTimestamp;
    private long id;
    private ArrayList<SegmentPoint> segmentPoints = new ArrayList<SegmentPoint>();

    public Segment(Date startedTimestamp) {
        this.startedTimestamp = startedTimestamp;
    }

    public Segment(long id, Date startedTimestamp) {
        this.id = id;
        this.startedTimestamp = startedTimestamp;
    }

    public Date getStartedTimestamp() {
        return startedTimestamp;
    }

    public long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public ArrayList<SegmentPoint> getSegmentPoints() {
        return segmentPoints;
    }

    public void setSegmentPoints(ArrayList<SegmentPoint> segmentPoints) {
        this.segmentPoints = segmentPoints;
    }
}

File: app/src/main/java/com/greenwald/aaron/ridetracker/model/Segment.kt
package com.greenwald.aaron.ridetracker.model

import java.io.Serializable
import java.util.Date

data class Segment(
    val id: SegmentId = 0,
    val startedTimestamp: Date,
    val stoppedTimestamp: Date? = null,
    val distance: Kilometers = Kilometers(0.0),
    val elapsedTime: Milliseconds = Milliseconds(0),
    val maxSpeed: KilometersPerHour = KilometersPerHour(0.0),
    val maxAltitude: Meters = Meters(0.0),
    val minAltitude: Meters = Meters(0.0),
    val segmentPoints: List<SegmentPoint> = emptyList()
): Serializable {
    val averageSpeed: KilometersPerHour get() = KilometersPerHour.from(distance, elapsedTime)
    val altitudeRange: Meters get() = Meters(maxAltitude.value - minAltitude.value)
}